### PR TITLE
Support `from_lists` for Tabular Classification and Regression (lists of dict/tuple)

### LINKED ISSUE #1326: Load tabular classifcation / regression data from lists of dicts
### Motivation

Currently in-memory data is limited to be a pandas dataframe

### Pitch

Add a method like the `from_lists` we have for text which would let you provide a list of dictionaries containing the data

### Alternatives

Force users to create a dataframe

### PR DESCRIPTION
**Pull request by krshrimali**
## What does this PR do?

This PR adds support for `from_lists` for tabular classification and regression. Doc artifact links:

1. [`TabularClassificationData.from_dicts`](https://output.circle-artifacts.com/output/job/8c1a3173-30df-4e51-a8a8-a27265ce6659/artifacts/0/html/api/generated/flash.tabular.classification.data.TabularClassificationData.html#flash.tabular.classification.data.TabularClassificationData.from_dicts)
2. [`TabularClassificationData.from_lists`](https://output.circle-artifacts.com/output/job/8c1a3173-30df-4e51-a8a8-a27265ce6659/artifacts/0/html/api/generated/flash.tabular.classification.data.TabularClassificationData.html#flash.tabular.classification.data.TabularClassificationData.from_lists)
3. [`TabularRegressionData.from_dicts`](https://output.circle-artifacts.com/output/job/8c1a3173-30df-4e51-a8a8-a27265ce6659/artifacts/0/html/api/generated/flash.tabular.regression.data.TabularRegressionData.html#flash.tabular.regression.data.TabularRegressionData.from_dicts)
4. [`TabularRegressionData.from_lists`](https://output.circle-artifacts.com/output/job/8c1a3173-30df-4e51-a8a8-a27265ce6659/artifacts/0/html/api/generated/flash.tabular.regression.data.TabularRegressionData.html#flash.tabular.regression.data.TabularRegressionData.from_lists)

Fixes #1326

## Before submitting
- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the **[contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md)**, Pull Request section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? [not needed for typos/docs]
- [x] Did you verify **new and existing tests pass** locally with your changes?
- [x] If you made a notable change (that affects users), did you **update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)**?

## PR review
 - [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/tabular/regression/data.py [hunk 1]
```diff
@@ -403,3 +404,114 @@ def from_dicts(
...
+            >>> train_data = [
+            ...     {"animal": "cat", "friendly": "yes", "weight": 6},
+            ...     {"animal": "dog", "friendly": "yes", "weight": 10},
+            ...     {"animal": "cat", "friendly": "no", "weight": 5},
+            ... ]
+            >>> predict_data = [
+            ...     {"friendly": "yes", "weight": 7},
+            ...     {"friendly": "no", "weight": 12},
+            ...     {"friendly": "yes", "weight": 5},
+            ... ]
```
Comment (ethanwharris) [inline flash/tabular/regression/data.py:471]: This block won't be visible in the rendered docs. You could either change it to just a `doctest` block or jsut provide the data literals when creating the datamodule below. Same applies to the classification docstring 😃

Comment (krshrimali) [reply flash/tabular/regression/data.py:490]: Ah, good observation! I just pushed commit to fixing it for `from_dicts` and `from_lists` (previous PR as well). I'll check with the rendered doc once CI is green.

Comment (krshrimali) [reply flash/tabular/regression/data.py]: This seems to be fixed now. Please take a look at the PR description for relevant links to the docs :)

**Review by ethanwharris — APPROVED** [review]
Awesome :smiley: LGTM!

**Pull request merged**